nestedtensor/nested/nested.py

```diff
@@ -294,8 +294,12 @@ def to_tensor_mask(self, mask_dim=None):
         element. These two tensors can be used to contruct a NestedTensor, however,
         nested_dim will be lost in this process."""
 
-        return masking.make_tensor_mask(self.to_list(), mask_dim)
+        # scalar
+        if not isinstance(self, list) and self.size() == (1,):
+            return masking.get_tensor_mask_tuple(self, mask_dim)
+            
+        return masking.get_tensor_mask_tuple(self.to_list(), mask_dim)
```

Comment:
This shouldn't be necessary anymore now that NestedTensor has support for __getitem__